test/sharness/lib/test-lib.sh

```diff
@@ -45,7 +45,9 @@ test "$TEST_EXPENSIVE" = 1 && test_set_prereq EXPENSIVE
 test "$TEST_NO_DOCKER" != 1 && type docker && test_set_prereq DOCKER
 
 TEST_OS=$(uname -s | tr [a-z] [A-Z])
-test expr "$TEST_OS" : "CYGWIN_NT" >/dev/null && test_set_prereq POSIX
+
+# Set a prereq as error messages are often different on Windows/Cygwin
+expr "$TEST_OS" : "CYGWIN_NT" >/dev/null || test_set_prereq STD_ERR_MSG
```

Comment:
Previously in #2368 you said "is the intent to not set POSIX when running on Cygwin? I thought Cygwin was POSIX-compliant."

And you are not wrong because on the Cygwin web site they say: "The core part is the Cygwin library which provides the POSIX system calls and environment these programs expect."

So just "POSIX" might make people think, like you did, that Cygwin should be included in this prereq. That's why I changed the name to something specific to error messages.